src/solve.jl

```diff
@@ -78,6 +78,18 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,
         error("You cannot use an DAE Algorithm with a ODEProblem")
     end
 
+    partitioned_ode_solvers = [Nystrom4(), IRKN3(), IRKN4(), ERKN4(), ERKN5(), ERKN7(), Nystrom4VelocityIndependent(),
+                               Nystrom5VelocityIndependent(), DPRKN4(), DPRKN5(), DPRKN6(), DPRKN6FM(), DPRKN8(), DPRKN12(),
+                               SymplecticEuler(), VelocityVerlet(), VerletLeapfrog(), PseudoVerletLeapfrog(), McAte2(), Ruth3(),
+                               McAte3(), CandyRoz4(), McAte4(), CalvoSanz4(), McAte42(), McAte5(), Yoshida6(), KahanLi6(),
+                               McAte8(), KahanLi8(), SofSpa10()]
```

Comment:
Use a subtype instead

Reply:
I am not sure if I am getting it right, so we need to specify a type of algorithms like ```abstract OrdinaryDiffEqPartitionedAlgorithms <: OrdinaryDiffEqAlgorithms end```, and then redefine all of the partitioned ODE methods like ```struct Nystrom4 <: OrdinaryDiffEqPartitionedAlgorithms end```?

Reply:
Yes, otherwise that type check allocates and is not inferred.